src/__tests__/userRouter/controller.test.ts

```diff
@@ -0,0 +1,342 @@
+import { Request, Response } from 'express';
+import { userService } from '@app/services';
+import * as controllers from '../../routers/userRouter/controllers';
+import { v4 } from 'uuid';
+import { makeShortId } from '@app/utils';
+import { CustomError } from '@app/errors';
+import { User } from '@app/models';
+
+const mockUsers = [
+  {
+    login: 'login',
+    password: 'password',
+    age: 20
+  }
+];
+
+jest.mock('../../services/userService');
+
+const mockNextFunction = jest.fn();
+
+const mockRequest = (
+  params: Partial<
+    Request<Record<string, any>, any, any, any, Record<string, any>>
+  >
+) => params;
+
+let responseObj = {};
+
+const mockQueries = [
+  {
+    params: {
+      query: {}
+    },
+    description: {
+      login: 'not passed',
+      limit: 10
+    },
+    result: {
+      loginSubstring: undefined,
+      limit: 10
+    }
+  },
+  {
+    params: {
+      query: {
+        loginSubstring: 'testLogin',
+        limit: '15'
+      }
+    },
+    description: {
+      login: 'testLogin',
+      limit: '15'
+    },
+    result: {
+      loginSubstring: 'testLogin',
+      limit: 15
+    }
+  },
+  {
+    params: {
+      query: {
+        limit: '12'
+      }
+    },
+    description: {
+      login: 'not passed',
+      limit: '12'
+    },
+    result: {
+      loginSubstring: undefined,
+      limit: 12
+    }
+  }
+];
```

Comment:
Maybe it'd be better to have some map/object with names. Only you know what each item in the array is used for, or I can check the usage, but readability is bad.